galley/pkg/runtime/processor.go

```diff
@@ -62,24 +82,46 @@ type postProcessHookFn func()
 
 // NewProcessor returns a new instance of a Processor
 func NewProcessor(src Source, distributor publish.Distributor, cfg *Config) *Processor {
-	state := newState(sn.DefaultGroup, metadata.Types, cfg, publish.NewStrategyWithDefaults(), distributor)
-	return newProcessor(state, src, nil)
+	stateStrategy := publish.NewStrategyWithDefaults()
+	stateListener := processing.ListenerFromFn(func(c resource.Collection) {
+		// When the state indicates a change occurred, update the publishing strategy
+		stateStrategy.OnChange()
+	})
+	state := newState(metadata.Types, cfg, stateListener)
+
+	// Use a more frequent publishing strategy for synthetic service entries.
+	serviceEntryStrategy := publish.NewStrategy(serviceEntryMaxWaitDuration, serviceEntryQuiesceDuration, serviceEntryTimerFrequency)
+	serviceEntryListener := processing.ListenerFromFn(func(c resource.Collection) {
+		// When the handler indicates a change occurred, update the publishing strategy
+		serviceEntryStrategy.OnChange()
+	})
+	serviceEntryHandler := serviceentry.NewHandler(cfg.DomainSuffix, serviceEntryListener)
+
+	return newProcessor(src, state, stateStrategy, serviceEntryHandler, serviceEntryStrategy, distributor, nil)
 }
 
 func newProcessor(
-	state *State,
 	src Source,
```

Comment:
I think this method is becoming too big, in terms of methods being passed in. Can't we create most of these in newProcessor? The whole idea with newProcessor is to make testing easier.

Reply:
it's been reduced since your original comment.